proto/penumbra/penumbra/core/transaction/v1/transaction.proto

```diff
@@ -114,6 +114,12 @@ message TransactionPerspective {
   repeated asset.v1.Metadata denoms = 5;
   // The transaction ID associated with this TransactionPerspective
   txhash.v1.TransactionId transaction_id = 6;
+  // A map of base64-encoded nullfiers to the IDs of the transactions containing
+  // the outputs that they nullify.
+  map<string, txhash.v1.TransactionId> transaction_ids_by_nullifier = 7;
+  // A map of base64-encoded commitments to the IDs of the transactions
+  // containing those commitments.
+  map<string, txhash.v1.TransactionId> transaction_ids_by_commitment = 8;
```

Comment:
Reviewers, please let me know what you think of this approach. It's the only way I could think of to easily allow views with links to other transactions to be generated from perspectives.

In the future, we'll also need to figure out a way to stuff output 1 and output 2 from the swap claim into the `TransactionPerspective` of a swap. But this work has already been taking me quite a while, so I wanted to at least get this merged first.

Reply:
Two comments:

1. I think the comment should read "transactions creating the commitments that they nullify" -- this is a minor nit but it's more technically correct since a swap commitment is a different kind of commitment than an output note

2. Why do we do base64 encoding of the nullifiers or commitments? This would force all of the consumers to add in special base64 parsing to interpret the perspective, right? Could we instead encode a list of `(Nullifier, TransactionId)` or `(Commitment, TransactionId)` pairs?

Reply:
I think there are restrictions on what types Proto3 allows to be used in maps -- like they have to be scalar types (?) -- so this would mean defining two sub-messages, each modeling one of the pairs, and then having a `repeated PairType` (which is basically the way that a `map` will be encoded on the wire anyways). Then we'd change the domain type to have a `BTreeMap<A, B>` and deserialize the proto-gen-type's `Vec<PairType>` into `BTreeMap<A, B>` using `.collect`.